core/src/main/java/ai/timefold/solver/core/impl/move/streams/dataset/AbstractUniDataStream.java

```diff
@@ -83,7 +80,7 @@ public UniDataStream<Solution_, A> addNull() {
         throw new UnsupportedOperationException();
     }
 
-    public AbstractDataset<Solution_, UniTuple<A>> createDataset() {
+    public UniDataset<Solution_, A> createDataset() {
```

Comment:
Should we avoid the use of specialized classes here?

Reply:
Makes sense to me this way - it's a uni stream, therefore it can afford to return a uni dataset. No need to cast it later.